src/pkrd/rng/mt.rs

```diff
@@ -7,7 +8,7 @@ impl MT {
     pub fn new(seed: u32) -> MT {
         let mut rng = MT {
             mt: [0; 624],
-            index: 624,
+            index: 0,
```

Comment:
to mimic the game, index would need to start at 0 as the original values are not twisted [see here](https://github.com/Admiral-Fish/CitraRNG/blob/59c6a2b5aaaf9eab7cf23386a5671795958d5f06/CitraRNG/mt.py#L6), if we start at 624 it will automatically shuffle and thus not predict the rng properly

Reply:
Oh good catch - looks like I was mistaken.  I think I was confusing two things when I wrote this.

In that case, how come `MT::default` has an index of 624 instead of 0?

Reply:
that was from me forgetting to also update default, ill update as mentioned here